src/fix_boundary_sources.cpp

```diff
@@ -0,0 +1,176 @@
+#include "meep_internals.hpp"
+#include "config.h"
+
+#ifdef HAVE_MPI
+#ifdef NEED_UNDEF_SEEK_FOR_MPI
+// undef'ing SEEK_* is needed for MPICH, possibly other MPI versions
+#undef SEEK_SET
+#undef SEEK_END
+#undef SEEK_CUR
+#endif
+#include <mpi.h>
+#include <algorithm>
+#endif
+
```

Comment:
You are only doing `#include <algorithm>` on MPI builds (it is inside the `#ifdef HAVE_MPI`) — you have to do it on serial builds too.

```suggestion
#include <mpi.h>
#endif

#include <algorithm>

```